File: tests/fixtures/testproject/swift/Engine.swift
class Base {}
protocol Store {}
class Settings {}
class Request {}

@available(*, deprecated)
class Engine: Base, Store {
  var settings: Settings
  init(settings: Settings) { self.settings = settings }
  
  @available(*, deprecated)
  func run(req: Request) -> Settings {
    let _ = try? String(contentsOfFile: "config/app.json")
    return Settings()
  }
}

File: tests/fixtures/testproject/dart/engine.dart
import 'dart:io';

class Base {}
mixin Store {}
class Settings {}
class Request {}

@Deprecated('demo')
class Engine extends Base with Store {
  Settings settings = Settings();
  
  @override
  Settings run(Request req) {
    final cfg = File('config/app.json');
    return Settings();
  }
}
